packages/sdk-react/src/components/userRoles/UserRoleDetails/UserRoleDetailsDialog/UserRoleDetailsDialog.tsx

```diff
@@ -0,0 +1,448 @@
+              };
+            }
+
+            return null;
+          })
+          .filter(
+            (
+              permission
+            ): permission is
+              | UserRoleCommonPermissions
+              | UserRolePayablePermissions => permission !== null
+          ),
+      },
+    };
+
+    const roleId = role?.id;
+
+    /*
+     * Generated types UpdateRoleRequest & CreateRoleRequest don't describe the real payload
+     * due to limitations of the current type generation library.
+     * We have to cast formattedData to UpdateRoleRequest to avoid type errors.
+     * The problem should be fixed when the sdk will use the Qraft library.
+     * */
+    if (roleId) {
+      return updateRole(roleId, formattedData as unknown as UpdateRoleRequest);
+    }
+
+    return createRole(formattedData as unknown as CreateRoleRequest);
+  };
+
+  const handleCancel = () => {
+    if (roleId) {
+      setView(UserRoleDetailsView.Read);
+
+      reset();
+
+      return;
+    }
+
+    if (dialogContext?.isDialogContent) {
+      dialogContext.onClose?.();
+    }
+  };
+
+  return (
+    <>
+      <DialogTitle>
+        <Box
+          display="flex"
+          justifyContent="space-between"
+          alignItems="center"
+          gap={2}
+        >
+          <Typography variant="h3" sx={{ wordBreak: 'break-word' }}>
+            {role
+              ? view === UserRoleDetailsView.Read
+                ? role.name
+                : t(i18n)`Edit User Role`
+              : t(i18n)`Create User Role`}
+          </Typography>
+          {dialogContext?.isDialogContent && (
+            <IconButton
+              edge="start"
+              color="inherit"
+              onClick={dialogContext.onClose}
+              aria-label={t(i18n)`Close role details`}
+            >
+              <CloseIcon />
+            </IconButton>
+          )}
+        </Box>
+      </DialogTitle>
+      <Divider />
+      <StyledDialogContainer>
+        <FormProvider {...methods}>
+          <form
+            id={formName}
+            noValidate
+            onSubmit={handleSubmit(handleRoleFormSubmission)}
+          >
+            {view == UserRoleDetailsView.Edit && (
+              <RHFTextField
+                label={t(i18n)`Name`}
+                name="name"
+                control={control}
+                fullWidth
+                required
+              />
+            )}
+            <StyledTableTitle variant="subtitle2" mt={4} mb={1}>
+              {t(i18n)`Permissions`}
+              <StyledDocLink
+                underline="none"
+                rel="noopener noreferrer"
+                href="https://docs.monite.com/docs/monite-account-structure#connect-entity-users"
+                target="_blank"
+              >
+                {t(i18n)`Go to Docs`}
+                &nbsp;
+                <OpenInNewIcon />
+              </StyledDocLink>
+            </StyledTableTitle>
+            <StyledTableContainer>
```

Comment:
I wanted to discuss this: don't you think `StyledTableContainer` only adds cognitive load (since it introduces a new entity)?
Don't you think that a simple:
```tsx
<TableContainer sx={{ minHeight: 300 }}>
```
easier to understand, whereas `StyledTableContainer` can hide a lot of things behind, and you will have to check its code first.

Reply:
I did this to maintain the same approach with styles because of many styled components in the current file. Sometime I also use inline styles if it's just one-two lines in one-two elements.

`StyledTableContainer` doesn't bother me since I am used to this pattern